### ISSUE TITLE
FirstOrError doesn't work with single source

### ISSUE THREAD

**Issue by diklimchuk** [badoo/Reaktive#153]
Sample test:

```
@Test
    fun firstOrErrorTest() {
        val observer = singleOf(2)
            .asObservable()
            .firstOrError()
            .test()
        assertEquals(2, observer.value)
    }
```
Works fine with observable.
The source of error is probably `firstOrAction` which emits error when `onComplete` is called.

Also, do you welcome prs?

**Comment by arkivanov** [comment]
Hey! The provided test passes. Are you using the `beta1` version?
PS: PRs are always welcome.

**Comment by arkivanov** [comment]
Please provide the failing case

**Comment by diklimchuk** [comment]
Yeah, the version is 1.0.0-beta1. I'll try to create a repo with this error when i have time.

**Comment by arkivanov** [comment]
Also could you please elaborate the problem as it is not clear. There is no such function 'Single.firstOrError' as it does not make sense, 'Single' always completes with either single value on error.

**Comment by diklimchuk** [comment]
No, the function is `Observable.firstOnError`, but it doesn't work only in case when `Single` is the actual source. So when i run `observableOf(1).firstOrError()` it executes successfully, but `singleOf(1).asObservable().firstOrError()` doesn't

**Comment by arkivanov** [comment]
Well, the provided test cases works. Waiting for failing test case.

**Comment by arkivanov** [comment]
Alright, the `1.0.0-beta1` has the bug, but it is already fixed in the latest master. Please wait for the release, it will be soon.

**Comment by arkivanov** [comment]
Fixed in a767bafb0c85bcd936543d7ed511ea73dc41a79e

**Comment by diklimchuk** [comment]
Great!
